.github/workflows/build.yml

```diff
@@ -0,0 +1,37 @@
+name: build
+on: [push, pull_request]
+jobs:
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        ruby: [3.2, 3.1, "3.0"]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            tmp
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+      - run: bundle exec rake compile
```

Comment:
I think actually this should be `bundle exec rake build` which would catch what appears to be a real issue in early ruby versions. See #1. I think to fix that I'll need to switch

```ruby
 spec.extensions = ["ext/tiktoken_ruby/Cargo.toml"]
 ```
 
 to 
 
 ```ruby
 spec.extensions = ["ext/tiktoken_ruby/extconf.rb"]
 ```